src/Frontend/TableHelper/Animation.js

```diff
@@ -24,227 +19,258 @@ export const stopStatus = { // 暫停時的狀態
-                    stopStatus.search = [count, i]
-                    sysStatusFunction()
-                    clearInterval(searchAnimation)
-                    return
+class searchAnimation_path extends searchAnimation_pos{
+    getItems(path, pathDirection){
+        this.items =  (path.length > 0) ? [path.shift(), pathDirection.shift()] : [undefined]
+    }
 
-                }else{
-                    if (WhichComponentSame(bomb[count][i]) > 3) {
-                        setTable(bomb[count][i], componentKind.search)
-                    }else{
-                        setTable(bomb[count][i], StartEndBombWeight(WhichComponentSame(bomb[count][i]), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearch))
-                    }
+    getClassName(id, d){
+        if (WhichComponentSame(id) > 3){
+            this.className1 = componentKind.path
+            this.className2 = direction(d)
+        } else {
+            this.className1 = componentKind.path
+            this.className2 = StartEndBombWeight(WhichComponentSame(id), direction(d), direction(d), componentKind.bombPath, componentKind.weightPath)
+        }
+    }
 
-                }
+    runAnimation(container){
+        var path = container[0];
+        var pathDirection = container[1];
 
-            }
+        this.getItems(path, pathDirection)
+
+        this.id = this.items[0]
+        if(this.id === undefined || path.length === 0){
+            return false
         }
-        count += 1
-    }, speed)
+        
+        this.newid = path[0]
+        this.d = pathDirection[0]
+
+        this.getClassName(this.newid, this.d)
+        setTable((WhichComponentSame(this.id) >= 3) ? this.id : -1, this.className1)
+        setTable(this.newid, this.className2)
+        return false
+    }
+    
 }
 
-export function PathAnimation(path, speed, pathDirection, sysStatusFunction, updateFunction) {
-    var [id, newid] = [stopStatus.pathID[0], stopStatus.pathID[1]]
-    var count = stopStatus.path
+function getAnimationObj(){
+    var animationObj
+    if(stopStatus.search.length > 0){
+        animationObj = [new searchAnimation_search(), stopStatus.search];
+    } else if (stopStatus.bomb.length > 0){
+        animationObj = [new searchAnimation_bomb(), stopStatus.bomb];
+    } else if (stopStatus.path.length > 0){
+        animationObj = [new searchAnimation_path(), [stopStatus.path, stopStatus.pathDirection]];
+    } else {
+        animationObj = [undefined, undefined];
+    }
+    return animationObj
+}
 
-    const pathAnimation = setInterval(() => {
-        if (count === path.length) {
-            resetAnimation()
-            updateFunction()
-            clearInterval(pathAnimation)
+export function SearchAnimation(search, bomb, path, pathDirection, speed, sysStatusFunction, updateFunction) {
+    if(stopStatus.complete){
+        path.unshift(-1)
+        pathDirection.unshift("")
+    
+        // I don't know why ???
```

Comment:
Please leave meaningful comments. Explain the situation or remove this comment.

Reply:
I use the "setSearchAnimation" (remove) to set the value of stopStatus.search、stopStatus.path...., but they not work, so I use line 169 ~ 173 to set the stopStatus and solve the problem

Reply:
Please remove unmeaningful comments. 
* [Line167](https://github.com/novayo/PathFinding/pull/1/files#diff-2f9bfd81a8b8ac2a6d4eccc01d42d1ec6c01c6bfdb30f8ef332cf93f899fe126R167)
* [Line168](https://github.com/novayo/PathFinding/pull/1/files#diff-2f9bfd81a8b8ac2a6d4eccc01d42d1ec6c01c6bfdb30f8ef332cf93f899fe126R168)